flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -260,6 +263,189 @@
 					text("\"ip\" - uses host's ip address as binding address"))
 				.build());
 
+	// ------------------------------------------------------------------------
+	//  Memory Options
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Total Process Memory size for the TaskExecutors.
+	 */
+	public static final ConfigOption<String> TOTAL_PROCESS_MEMORY =
+		key("taskmanager.memory.total-process.size")
+			.noDefaultValue()
+			.withDescription("Total Process Memory size for the TaskExecutors. This includes all the memory that a"
+				+ " TaskExecutor consumes, consisting of Total Flink Memory, JVM Metaspace, and JVM Overhead. On"
+				+ " containerized setups, this value is automatically configured to the size of the container.");
+
+	/**
+	 * Total Flink Memory size for the TaskExecutors.
+	 */
+	@Documentation.CommonOption(position = Documentation.CommonOption.POSITION_MEMORY)
+	public static final ConfigOption<String> TOTAL_FLINK_MEMORY =
+		key("taskmanager.memory.total-flink.size")
```

Comment:
Maybe

```suggestion
		key("taskmanager.memory.flink.size")
```

Reply:
Maybe not, if `flink` alone could be confused with `framework`.

Reply:
Same here.

Reply:
ok